src/components/AlphabetFilter/AlphabetFilter.tsx

```diff
@@ -0,0 +1,197 @@
+      { text: "Q", value: "q" },
+      { text: "R", value: "r" },
+      { text: "S", value: "s" },
+      { text: "T", value: "t" },
+      { text: "U", value: "u" },
+      { text: "V", value: "v" },
+      { text: "W", value: "w" },
+      { text: "X", value: "x" },
+      { text: "Y", value: "y" },
+      { text: "Z", value: "z" },
+      { text: "Show All", value: "showAll" },
+    ];
+
+    const refCurrentLetter = useRef(currentLetter);
+    const [selectedLetter, setSelectedLetter] = useState<string>(currentLetter);
+
+    // If the parent passes down a new currentLetter, and an onClick
+    // function exists, then set the internal state – selectedLetter –
+    // to the new currentLetter and update the refCurrentLetter with that value.
+    React.useEffect(() => {
+      if (
+        onClick &&
+        currentLetter &&
+        currentLetter !== refCurrentLetter.current
+      ) {
+        setSelectedLetter(currentLetter);
+        refCurrentLetter.current = currentLetter;
+      }
+    }, [currentLetter, onClick]);
+
+    /**
+     * This function is only called when clicking on a link should
+     * not update the URL or refresh the page.
+     */
+    const handleOnClick = (
+      e: React.MouseEvent | React.KeyboardEvent,
+      clickedLetter: string
+    ) => {
+      e.preventDefault && e.preventDefault();
+      if (selectedLetter === clickedLetter) return;
+      setSelectedLetter(clickedLetter);
+      onClick && onClick(clickedLetter);
+    };
+
+    const getButtonElement = (item) => {
+      const isSelectedLetter =
+        selectedLetter === item.value && item.value !== "showAll";
+      const isShowAll = item.value === "showAll";
+      const isButtonDisabled =
+        isDisabled ||
+        (activeLetters && !activeLetters.includes(item.value) && !isShowAll);
+
+      const buttonStyles = isSelectedLetter
+        ? { ...styles, border: "1px solid gray" }
+        : isShowAll
+        ? {
+            ...styles,
+            fontWeight: "normal !important",
+            whiteSpace: "nowrap",
+            marginLeft: "var(--nypl-space-m)",
+          }
+        : styles;
+
+      return (
+        <Button
+          aria-current={isSelectedLetter ? "page" : null}
+          aria-label={
+            item.text === "Show All" ? item.text : "Page " + item.text
+          }
+          buttonType="link"
+          id={`filter-${item.value}`}
+          isDisabled={isButtonDisabled}
+          __css={buttonStyles}
+          onClick={(e: React.MouseEvent | React.KeyboardEvent) => {
+            handleOnClick(e, item.value);
+          }}
+        >
+          {item.text}
+        </Button>
+      );
+    };
+
+    const getFilterLetters = () => {
+      if (isLargerThanMobile) {
+        return filterButtons.map((item) => {
+          return getButtonElement(item);
+        });
+      }
+      // split letters into rows of 10 + Show All
+      const mobileRows = filterButtons.reduce((rows, key, index) => {
```

Comment:
This makes sense when reading through the code but can you add some comments to make it quicker to read through?

Reply:
Also making it clear that the returned value is a 2d array.